src/controllers/teamController.ts

```diff
@@ -0,0 +1,360 @@
+    }
+    
+    const team = await teamRepository.findOne({uuid: teamId});
+
+    if(team){
+        team.name = newName;
+        await teamRepository.save(team);
+        res.sendStatus(200);
+    } else {
+        res.sendStatus(401);
+    }
+    
+
+}
+
+export const joinTeam = async (req: Request, res: Response): Promise<void> => {
+    const teamRepository = getManager().getRepository(Team);
+    const inviteRepo = getManager().getRepository(TeamInvite);
+
+    const teamId = req.params.teamId;
+    const inviteId = req.body.inviteId;
+    const token = req.header('Authorization')?.split(' ')[1];
+    if(!token) {
+        res.sendStatus(401); // User was not properly authenticated...
+        return;
+    }
+
+    const user = await auth.getUserObjectFromToken(token, ['team', 'teamInvites']);
+    if(!user) {
+        res.sendStatus(401);
+        return;
+    }
+    
+    if(user.team) {
+       res.status(400).send('This user is already on a team.');
+       return;
+    }
+
+    const invite = user.teamInvites?.find(inv => inv.uuid === inviteId);
+    if(!invite) {
+        res.status(403).send('You are not invited to this team.');
+        return;
+    }
+
+    const team = await teamRepository.findOne({uuid: teamId});
+    if(!team){
+        res.status(404).send('That team does not exist.');
+        return;
+    }
+
+    if(team.members && team.members.length >= MAX_TEAM_SIZE) {
+        res.status(400).send('That team is full.');
+        return;
+    }
+    
+    delete user.team;
+    if(team.members){
+        team.members.push(user);
+    } else {
+        team.members = [user];
+    }
+    
+    await inviteRepo.remove(invite);
+    await teamRepository.save(team);
+    res.sendStatus(200);
+}
+
+export const leaveTeam = async (req: Request, res: Response): Promise<void> => {
+    const teamRepository = getManager().getRepository(Team);
+
+    const token = req.header('Authorization')?.split(' ')[1];
+    if(!token) {
+        res.sendStatus(401); // User was not properly authenticated...
+        return;
+    }
+
+    const user = await auth.getUserObjectFromToken(token, ['team']);
+    if(!user) {
+        res.sendStatus(401);
+        return;
+    }
+    
+    if(!user.team) {
+       res.status(400).send('This user is not on a team.');
+       return;
+    }
+
+    const team = await teamRepository.findOne({uuid: user.team.uuid});
+
+    if(!team){
+        res.status(404).send('That team does not exist???');
+        return;
+    }
+
+    // Remove the user
+    team.members.forEach((item, index) => {
+        if(item.uuid == user.uuid) team.members.splice(index, 1);
+    });
+
+    // Team is empty, delete it
+    if(team.members.length == 0) {
+        await teamRepository.remove(team);
+    } else {
+        await teamRepository.save(team);
+    }
+
+    res.sendStatus(200);
+}
+
+
+// Takes email or discord username
+export const createInvite = async (req: Request, res: Response): Promise<void> => {
```

Comment:
Just noting for future reference that any number of invites can be created, but validation of team size happens when joining a team.